modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java

```diff
@@ -598,6 +598,12 @@ public static boolean needCastInSearchBounds(IgniteTypeFactory typeFactory, RelD
             return false;
         }
 
+        // TIME, TIMESTAMP and TIMESTAMP_LTZ can use index, ignoring precision.
+        if (fromType.getSqlTypeName() == toType.getSqlTypeName()
```

Comment:
next cases are unclear for now : create table ... col TIME(3) and further query : select from .. where col = TIME '12:12:12.1234' is it need to be implicitly casted ? And on the contrary: create table ... col TIME(3) and further : select from .. where col = TIME '12:12:12.12'. Other words - temporal predicate with great or less precision than column is.